### PR TITLE
Use npm config when generating the npm lockfile

### PR DESCRIPTION
**Pull request by laurence79**
Great tool!

I've just started to use it within a corporate environment and found that when using npm as the package manager, the generate lock file stage can fail if the environment isn't using the default https://registry.npmjs.org repo.

This PR addresses that by loading the npm config, using [@npmcli/config](https://www.npmjs.com/package/@npmcli/config?activeTab=readme) and passing that to Arborist.

I've validated the approach using patch-package in the environment in question.

### CODE REVIEW

File: src/lib/lockfile/helpers/load-npm-config.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Config from "@npmcli/config";
+import defaults from "@npmcli/config/lib/definitions/index.js";
+
+export async function loadNpmConfig({ npmPath }: { npmPath: string }) {
+  const conf = new Config({
```
Comment (0x80) [inline src/lib/lockfile/helpers/load-npm-config.ts]: Nitpick; I don't think saving these 2 characters is worth making things less readable. I typically avoid abbreviations

File: .DS_Store
Comment (0x80) [inline .DS_Store:1]: Seems like I forgot to add this to .gitignore. Could you please add it and remove this file?

**Review by 0x80 — CHANGES_REQUESTED** [review]
Thanks @laurence79 this looks great! If you could fix these minor things I'll try to publish it soon.

File: src/lib/lockfile/helpers/load-npm-config.ts (hunk 1 above)
Comment (laurence79) [reply src/lib/lockfile/helpers/load-npm-config.ts]: Understood, addressed.

File: .DS_Store
Comment (laurence79) [reply .DS_Store:1]: Slightly embarrassed that this one got committed 😳
Removed and `.gitignore` updated.

**Comment by laurence79** [comment]
@0x80 comments addressed, cheers!

**Review by 0x80 — APPROVED** [review]

**Pull request merged**